packages/fast-element/src/directives/when.ts

```diff
@@ -0,0 +1,81 @@
+import { DOM } from "../dom";
+import { ITemplate, ICaptureType } from "../template";
+import { ISyntheticView } from "../view";
+import { IExpression, AccessScopeExpression, Getter } from "../ast";
+import { IBehavior } from "../behaviors/behavior";
+import {
+    IPropertyChangeListener,
+    Observable,
+    IGetterInspector,
+} from "../observation/observable";
+import { BindingDirective } from "./bind";
+
+export class WhenDirective extends BindingDirective {
+    behavior = WhenBehavior;
+
+    constructor(public expression: IExpression, public template: ITemplate) {
+        super(expression);
+    }
+
+    public createPlaceholder(index: number) {
+        return DOM.createLocationPlaceholder(index);
+    }
+}
+
+export class WhenBehavior
+    implements IBehavior, IGetterInspector, IPropertyChangeListener {
+    private location: Node;
+    private view: ISyntheticView | null = null;
+    private chachedView?: ISyntheticView;
```

Comment:
Should this be `cachedView`?

Reply:
It's the past tense of the verb "to chacha".
Used in a sentence: "Did you see how @chrisdholt chached last night?"
When you see someone chacha, you have a "chached view". Also, if they are not latino, it's a "synthetic chached view".